test/test_model_inheritance.py

```diff
@@ -105,5 +105,8 @@ class Meta:
         class ChildModel(ModelWithIndexes):
             pass
 
+        # force connection
+        ModelWithIndexes._mongometa.collection
```

Comment:
The index creation happens when you access the `collection` property, instead of when the model class is created.  So without this line, the assertion about the index existing will fail.

Reply:
Ah I see. I think this should be updated to be a little more explicit:

```python
         index_info = DB.model_with_indexes.index_information()
         self.assertNotIn('product_id_1_name_1', index_info)

         # Indexes are only created once the Model's collection is accessed.
         ModelWithIndexes._mongometa.collection

         index_info = DB.model_with_indexes.index_information()
         self.assertTrue(index_info['product_id_1_name_1']['unique'])

         # Creating indexes on the child should not error.
         ChildModel._mongometa.collection

         # Index info should not have changed.
         final_index_info = DB.model_with_indexes.index_information()
         self.assertEqual(index_info, final_index_info)
```